tools/sigmut/sigmut.xml

```diff
@@ -0,0 +1,495 @@
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf5" label="--> SBS 6 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf6" label="--> SBS 24 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf7" label="--> SBS 96 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf8" label="--> SBS 384 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf9" label="--> SBS 1536 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+		<data format="txt" name="SeqInf10" label="--> SBS 6144 Sig. Mut. ALL">
+			<filter>set_analysis['choices'] == 'get_sigmut' and set_analysis['seqInfo'] is True</filter>
+		</data>
+
+
+	</outputs>
+
+	<tests>
+		<test>
+			<conditional name="set_analysis">
+				<param name="choices" value="install_genome"/>
+				<param name="refgendwn" ftype="fasta" value="GRCh37"/>
+			</conditional>
+			<output name="logref" file="GRCh37.log" lines_diff="5"/>
+		</test>
+
+		<test>
+			<conditional name="set_analysis">
+				<param name="choices" value="install_genome"/>
+				<param name="refgendwn" ftype="fasta" value="GRCh38"/>
+			</conditional>
+			<output name="logref" file="GRCh38.log" lines_diff="5"/>
+		</test>
+
+
+		<test>
+			<conditional name="set_analysis">
+				<param name="choices" value="get_sigmut"/>
+				<param name="refgendat" ftype="fasta" value="GRCh37"/>
+				<conditional name="vcfile_input">
+					<param name="vcfile" value="icgc"/>
+					<param name="icgc_file" ftype="txt" value="test_matrix.txt"/>
+				</conditional>
+				<conditional name="bed_input">
+					<param name="bedfile" value="no"/>	
+				</conditional>
+				<param name="plot" value="True"/>
+			</conditional>
+
+		<output name="SBS6" file="SBS_6.pdf" lines_diff="5"/>
+		<output name="SBS24" file="SBS_24.pdf" lines_diff="5"/>
+		<output name="SBS78" file="SBS_78.pdf" lines_diff="5"/>
+		<output name="SBS96" file="SBS_96.pdf" lines_diff="5"/>
+		<output name="SBS384" file="SBS_384.pdf" lines_diff="5"/>
+		<output name="SBS1536" file="SBS_1536.pdf" lines_diff="5"/>
+		<output name="DBS78" file="DBS_78.pdf" lines_diff="5"/>
+		<output name="DBS186" file="DBS_186.pdf" lines_diff="5"/>
+
+		</test>
+
+
+		<test>
+                        <conditional name="set_analysis">
+                                <param name="choices" value="get_sigmut"/>
+                                <param name="refgendat" ftype="fasta" value="GRCh37"/>
```

Comment:
To clarify. the Reference data are installed/downloaded when the conda package is installed ? or when the tool is running ?

Reply:
The reference genome(s) is(are) downloaded before calculating the mutational signatures. It is a mandatory step, because without these files the tool can not be run any further.
However, we can skip this step/option by including already downloaded files, but then the conda package will be extremely large in size. In addition, we will have to update the package very frequently for newer versions of these genome files.

Reply:
OK, I see. Thus my guess is that the failing is travis test is a time-out during this particular step, not during the conda package install.

This is actually a good news :  
Would it be possible to create a distant repo for a very slight genome prepared in silico ? The benefit from this not only being actual automated testing in travis (to which I am strongly attached as you know) but also to mastering (++ autonomy) the format of reference for the alexandrov et al tool !

Reply:
Unfortunately, the reference genomes are provided by the authors as already compiled files. The script downloads the information from the following link:
ftp://ngs.sanger.ac.uk/scratch/project/mutographs/SigProf/
However, there should be a super complex way of processing a particular genome, that includes not only the fasta file(s), but also annotating all gene coordinates. I am checking this possibility, but not easy at all.

Reply:
But in the meantime, ftp://ngs.sanger.ac.uk/scratch/project/mutographs/SigProf/c_elegans.tar.gz is exactly what we need ! then we just have to find a test with c. elegans datasets, and I predict that the test will rock !